packages/collector/test/tracing/database/db2/test.js

```diff
@@ -0,0 +1,1117 @@
+  return agentControls.getSpans().then(spans => {
+    if (options.expectSpans === false) {
+      expect(spans).to.be.empty;
+      return;
+    }
+
+    const entrySpan = testUtils.expectAtLeastOneMatching(spans, [
+      span => expect(span.n).to.equal('node.http.server'),
+      span => expect(span.f.e).to.equal(String(controls.getPid())),
+      span => expect(span.f.h).to.equal('agent-stub-uuid')
+    ]);
+
+    if (options.expectNoDb2Span) {
+      expect(spans.length).to.equal(1);
+      return;
+    }
+
+    expect(spans.length).to.equal(options.spanLength || 2);
+
+    if (options.verifyCustom) return options.verifyCustom(entrySpan, spans);
+
+    testUtils.expectAtLeastOneMatching(spans, [
+      span => expect(span.t).to.equal(entrySpan.t),
+      span => expect(span.p).to.equal(entrySpan.s),
+      span => expect(span.n).to.equal('db2'),
+      span => expect(span.k).to.equal(constants.EXIT),
+      span => expect(span.f.e).to.equal(String(controls.getPid())),
+      span => expect(span.f.h).to.equal('agent-stub-uuid'),
+      span => expect(span.data.db2.stmt).to.equal(options.stmt || 'select 1 from sysibm.sysdummy1'),
+      span => expect(span.data.db2.dsn).to.equal(`${CONN_STR};DATABASE=${DB2_NAME}`),
+      span => expect(span.async).to.not.exist,
+      span =>
+        options.error
+          ? expect(span.data.db2.error).to.contain(options.error)
+          : expect(span.data.db2.error).to.not.exist,
+      span => (options.error ? expect(span.ec).to.equal(1) : expect(span.ec).to.equal(0))
+    ]);
+  });
+};
+
+mochaSuiteFn('tracing/db2', function () {
+  const timeout = 30 * 1000;
+  this.timeout(timeout);
+
+  globalAgent.setUpCleanUpHooks();
+  const agentControls = globalAgent.instance;
+  let controls;
+
+  describe('tracing is active', function () {
+    before(async () => {
+      DB2_NAME = getDatabaseName();
+      TABLE_NAME_1 = generateTableName();
+      TABLE_NAME_2 = generateTableName();
+      TABLE_NAME_3 = generateTableName();
+
+      controls = new ProcessControls({
+        dirname: __dirname,
+        port: 3322,
+        useGlobalAgent: true,
+        env: {
+          DB2_NAME,
+          DB2_TABLE_NAME_1: TABLE_NAME_1,
+          DB2_TABLE_NAME_2: TABLE_NAME_2,
+          DB2_TABLE_NAME_3: TABLE_NAME_3
+        }
+      });
+
+      ProcessControls.setUpTestCaseCleanUpHooks(controls);
+      await controls.startAndWaitForAgentConnection();
+    });
+
+    after(async () => {
+      await controls.sendRequest({
+        method: 'DELETE',
+        path: '/tables'
+      });
+
+      await controls.sendRequest({
+        method: 'DELETE',
+        path: '/conn'
+      });
+
+      await controls.stop();
+    });
+
+    it('must trace query promise', function () {
+      return controls
+        .sendRequest({
+          method: 'GET',
+          path: '/query-promise'
+        })
+        .then(() => {
+          return testUtils.retry(() => {
+            return verifySpans(agentControls, controls);
```

Comment:
_suggestion/issue:_ In the app under test we already return the data from the query via the HTTP response, so we have it available here (as the resolved value of the `sendRequest` promise). It would be a great addition to generally check the HTTP response. The added benefit is that we do not only verify that the produced spans are correct, but also that our instrumentation code does not change or swallow result values.  IMO we should do that in every integration test.

Reply:
Ok will add the response body check 👍